code/game/objects/items/spear.dm

```diff
@@ -153,8 +153,14 @@
 	if(iseffect(AM)) //and no accidentally wasting your moment of glory on graffiti
 		return
 	user.say("[war_cry]", forced="spear warcry")
-	explosive.forceMove(AM)
-	explosive.detonate(lanced_by=user)
+	if(isliving(user))
+		var/mob/living/living_user = user
+		living_user.set_resting(TRUE, TRUE, TRUE)
```

Comment:
Named parameters

Reply:
we're not skipping any parameters here therefore they don't need to be explicitly named, we have autodoc for a reason

Reply:
You are wrong

https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#prefer-named-arguments-when-the-meaning-is-not-obvious